src/daft-plan/src/physical_planner/translate.rs

```diff
@@ -434,6 +434,7 @@ pub(super) fn translate_single_logical_node(
                 left_clustering_spec.num_partitions(),
                 right_clustering_spec.num_partitions(),
             );
+            let num_partitions = min(num_partitions, cfg.shuffle_join_default_partitions);
```

Comment:
I believe this logic here will only be used in sort merge join, so can we move it to that branch? Also, it should be a max here as well.